src/main/java/uk/gov/hmcts/juror/api/moj/service/jurormanagement/JurorAppearanceServiceImpl.java

```diff
@@ -328,6 +329,102 @@ public AttendanceDetailsResponse deleteAttendance(BureauJwtPayload payload, Upda
         return response;
     }
 
+    @Override
+    @Transactional
+    public void modifyConfirmedAttendance(ModifyConfirmedAttendanceDto request) {
+
+        final LocalDate attendanceDate = request.getAttendanceDate();
+        final String jurorNumber = request.getJurorNumber();
+        final String poolNumber = request.getPoolNumber();
+
+
+        log.debug(
+            String.format("User %s is modifying attendance for juror %s", SecurityUtil.getActiveLogin(), jurorNumber));
+
+        JurorPool jurorPool = jurorPoolRepository.findByJurorJurorNumberAndPoolPoolNumber(jurorNumber,
+            poolNumber);
+
+        if (jurorPool == null) {
+            throw new MojException.NotFound("No valid juror pool found for juror " + jurorNumber, null);
+        }
+
+        // validate the court user has access to the juror and pool
+        JurorPoolUtils.checkOwnershipForCurrentUser(jurorPool, SecurityUtil.getActiveOwner());
+
+        // get the appearance record if it exists
+        Appearance appearance =
+            appearanceRepository.findByJurorNumberAndPoolNumberAndAttendanceDate(jurorNumber, poolNumber,
+                attendanceDate).orElseThrow(() -> new MojException.NotFound("No valid appearance record found", null));
+
+        modifyConfirmedAttendance(appearance,
+            request.getModifyAttendanceType(),
+            request.getCheckInTime(),
+            request.getCheckOutTime());
+    }
+
+    private void modifyConfirmedAttendance(Appearance appearance,
+                                           ModifyConfirmedAttendanceDto.ModifyAttendanceType modifyAttendanceType,
+                                           LocalTime checkInTime,
+                                           LocalTime checkOutTime
+    ) {
+        boolean isLongTrail = jurorExpenseService.isLongTrialDay(
```

Comment:
Typo - isLongTrail should be isLongTrial

Reply:
updated spelling